### PR TITLE
OLS-225:  add unit tests

### PR DESCRIPTION
**Pull request by raptorsun**
## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes # [OLS-255](https://issues.redhat.com//browse/OLS-255)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: internal/controller/ols_app_server_assets_test.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+	Context("complete custome resource", func() {
+		BeforeEach(func() {
+			rOptions = &OLSConfigReconcilerOptions{
+				LightspeedServiceImage: "lightspeed-service:latest",
+			}
+			cr = getCompleteOLSConfigCR()
+			r = &OLSConfigReconciler{
+				Options:    *rOptions,
+				logger:     logf.Log.WithName("olsconfig.reconciler"),
+				Client:     k8sClient,
```
Comment (vishnuchalla) [inline internal/controller/ols_app_server_assets_test.go:38]: @raptorsun is this a fake client or real? Just trying to understand if these tests actually run on a real cluster or somewhere else.

Comment (raptorsun) [reply internal/controller/ols_app_server_assets_test.go:38]: Hello this client is a real client from `sigs.k8s.io/controller-runtime/pkg/client`. 
The test in this file does not use the client, it is just for asset generations. 
The test in the file `ols_app_server_reconciliator_test.go` checks the assests are accepted by kube api server and created/updated correctly. The test uses [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) that spins up an etcd server with a kubenetes api server just for the test

File: internal/controller/ols_app_server_assets_test.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+					Port:       OLSAppServerServicePort,
+					Protocol:   corev1.ProtocolTCP,
+					TargetPort: intstr.Parse("http"),
+				},
+			}))
+		})
+
+	})
+
+	Context("empty custome resource", func() {
```
Comment (xrajesh) [inline internal/controller/ols_app_server_assets_test.go:167]:
```suggestion
	Context("empty custom resource", func() {
```

**Pull request merged**